pkg/ephemeral/network/proxy.go

```diff
@@ -174,25 +187,36 @@ func RetryingDialer(sleep, timeout time.Duration, sideEffect func()) func(addr,
 }
 
 // RetryingDialerWithContext tries to establish a TCP connection to a socket until the timeout is reached or the context is cancelled.
-func RetryingDialerWithContext(sleep, timeout time.Duration, sideEffect func()) func(ctx context.Context, addr, port string) (conn net.Conn, err error) {
-	return func(ctx context.Context, addr, port string) (conn net.Conn, err error) {
+func RetryingDialerWithContext(sleep time.Duration, l *zap.SugaredLogger) func(ctx context.Context, addr, port string, timeout time.Duration) (conn net.Conn, err error) {
+	return func(ctx context.Context, addr, port string, timeout time.Duration) (conn net.Conn, err error) {
 		started := time.Now()
+		logTicker := time.NewTicker(5 * time.Second)
+		connectTimer := time.NewTimer(0)
+		defer logTicker.Stop()
+		defer connectTimer.Stop()
 		for {
 			select {
 			case <-ctx.Done():
-				return conn, errors.New("context cancelled")
-			default:
+				return conn, errors.New("cancelled connection attempt for %s:%s - context done")
```

Comment:
The variables for the substitution are missing here

Reply:
fixed